testing/test_skipping.py

```diff
@@ -235,6 +235,25 @@ def test_func2():
             ["*def test_func():*", "*assert 0*", "*1 failed*1 pass*"]
         )
 
+    @pytest.mark.parametrize(
```

Comment:
The test as written does not test the change, because it passes also before the change. A good regression test should fail without the change, otherwise it is testing something different.

Reply:
Thanks for your comments @bluetech 
I am trying to write an appropriate test but it seems when `tox -e py38 -- test_skipping.py` runs the test, the output is different when I run the test by myself with this command `pytest -rs --runxfail`  
There are no breaks in the location 

![Capture_TOX](https://user-images.githubusercontent.com/5881793/86511491-16c98280-bdfa-11ea-94c6-6e38c4d021f2.PNG)

And this is the result when I do it with `pytest` command.

![Capture_runxfail](https://user-images.githubusercontent.com/5881793/86511519-58f2c400-bdfa-11ea-9523-a6576857d98e.PNG)

could you please help me to understand the issue?

Reply:
The `pytest` in your second image is `pytest-5.4.3`, which does not include the fix, so is pointing at the internal `skipping.py:90`. This is what we *don't* want to happen. The first image contains the fix, so points to the correct `test_xfail_run_with_skip_mark.py:2`. What you should assert is what's in the first image.

Reply:
Oops! I forgot to mention that I removed the fix from my code and test it again to make sure it's going to break but it didn't as you can see in the first picture.
Then I tried to do it again without `tox` as you can see in the second picture.

Reply:
I'm not sure I understand, so I'll just say the problem with the test is that it checks 

```
*unconditional skip*
```

that first `*` means that the test accepts any path, which is what the PR changes, so it passes both before and after.

What you want to do is include the path in the test, so

```
*SKIPPED [1] test_xfail_run_with_skip_mark.py:2: unconditional skip
```

I'm not sure exactly if the testdir generates this file name, you may want to give an explicit name like this:

```py
p = testdir.makepyfile(
    test_it="..."
)
```

then the file name should be `test_it.py`.

Reply:
I totally understood what you meant in the first place. 
I changed it to check the same desired output(almost the same as you mentioned here in your last comment). then I removed my fix from `skipping.py` to make sure my test is going to fail without the fix. but it didn't. As I said before with provided screenshots, it seems when you use `tox` command and run your tests, the problem disappears and there is no need for the fix. 
BUT when you are using the normal way with `pytest -rs --runxfail` command for some sample file the problem shows itself. 
If I want to make it more clear you can put the test inside your `test_skipping.py` on your local (without adding the fix inside `skipping.py`) and see the result.
I think this test should logically satisfy our condition. It should fail without the fix and should pass with the fix.
```
@pytest.mark.parametrize(
       "test_input,expected",
        [
            ("-rs", ["SKIPPED [1] test_sample.py:2: unconditional skip", "*1 skipped*"]),
            ("-rs --runxfail", ["SKIPPED [1] test_sample.py:2: unconditional skip", "*1 skipped*"]),
        ],
    )
    def test_xfail_run_with_skip_mark(self, testdir, test_input, expected):
        testdir.makepyfile(
            test_sample="""
            import pytest
            @pytest.mark.skip
            def test_skip_location() -> None:
                assert 0
        """
        )
        result = testdir.runpytest(test_input)
        result.stdout.fnmatch_lines(expected)
``` 
I'm trying to find the reason for this weird behavior and fix it.

Reply:
Ah I understand now.

The test you pasted above has two issues:

1. Instead of making the `test_input` param a single string, it should be a list of strings instead (like `["-rs", "--runxfail"]` for the second case), and used like this: `result = testdir.runpytest(*test_input)` (notice the `*`).

    BTW, I would also call this param `args` instead of `test_input`.

2. Since `fnmatch` is used, the `[` and `]` in `SKIPPED [1]` have a special meaning, so need to be escaped like so: `SKIPPED [[]1[]]`. Looks a bit weird but should work.

Reply:
Thank you very much for your guidance. It helped me to finally fix the problem.